### PR TITLE
Lde/lookup grand prod relation

### PR DESCRIPTION
**Pull request by ledwards2225**
# Description

Add the lookup grand product relations for Ultra Honk. (Part of issue #342)

# Checklist:

- [ ] I have reviewed my diff in github, line by line.
- [ ] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [ ] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [ ] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [ ] There are no circuit changes, OR a cryptographer has been assigned for review.
- [ ] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [ ] The branch has been rebased against the head of its merge target.
- [ ] I'm happy for the PR to be merged at the reviewer's next convenience.
- [ ] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [ ] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

File: cpp/src/barretenberg/honk/composer/ultra_honk_composer.test.cpp [hunk 1]
```diff
@@ -39,34 +39,31 @@ std::vector<uint32_t> add_variables(auto& composer, std::vector<fr> variables)
...
 
-    // Check that sorted witness-table and table polys agree
+    // Check that sorted witness-table polynomials agree
     for (auto& entry : honk_store) {
         std::string key = entry.first;
         bool is_sorted_table = (key.find("s_") != std::string::npos);
-        bool is_table = (key.find("table_value_") != std::string::npos);
-        if (plonk_store.contains(key) && (is_sorted_table || is_table)) {
+        if (plonk_store.contains(key) && is_sorted_table) {
+            info(key);
```
Comment (Rumata888) [inline cpp/src/barretenberg/honk/composer/ultra_honk_composer.test.cpp]: Don't forget to delete the info

File: cpp/src/barretenberg/honk/sumcheck/relations/lookup_grand_product_relation.hpp [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+#include "relation.hpp"
+#include "barretenberg/honk/flavor/flavor.hpp"
+#include "../polynomials/univariate.hpp"
+
+namespace proof_system::honk::sumcheck {
+
+template <typename FF> class LookupGrandProductComputationRelation {
+  public:
+    // 1 + polynomial degree of this relation
+    static constexpr size_t RELATION_LENGTH = 4; // deg(z_lookup * wire_accum * q_lookup) = 3
```
Comment (Rumata888) [inline cpp/src/barretenberg/honk/sumcheck/relations/lookup_grand_product_relation.hpp]: Is it 4? It seems that we multiply many more selectors and wires

Comment (Rumata888) [reply cpp/src/barretenberg/honk/sumcheck/relations/lookup_grand_product_relation.hpp]: The wire accumulator is already a product of several wires

**Review by Rumata888 — COMMENTED** [review]
I think there might be an issue with the relation length estimate, otherwise everything seems fine.

File: cpp/src/barretenberg/honk/composer/ultra_honk_composer.test.cpp (hunk 1 above)
Comment (ledwards2225) [reply cpp/src/barretenberg/honk/composer/ultra_honk_composer.test.cpp]: Good catch, done.

File: cpp/src/barretenberg/honk/sumcheck/relations/lookup_grand_product_relation.hpp (hunk 2 above)
Comment (ledwards2225) [reply cpp/src/barretenberg/honk/sumcheck/relations/lookup_grand_product_relation.hpp]: The accumulators are just linear combinations of the component polynomials so they don't have a higher degree - but you're right, my degree was off by one. The highest degree terms are of the form `z_lookup * wire_accum * q_lookup * table` -- I was missing the table poly in my calculation. Good catch!

**Review by Rumata888 — APPROVED** [review]

**Pull request merged**